tests/test_jsvalidation.py

```diff
@@ -0,0 +1,419 @@
+# License, or (at your option) any later version.
+#
+# DoSchema is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoSchema; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+import jsonschema
+import pytest
+
+from doschema.errors import JSONSchemaCompatibilityException, \
+    JSONSchemaDependenciesException
+from doschema.jsvalidation import JSONSchemaValidation
+
+
+def test_special_field_fail():
+    v1 = {
+        "type": "object",
+        "oneOf": [{
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "string"}
+            }
+        }, {
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "integer"}
+            }
+        }]
+    }
+    obj = JSONSchemaValidation()
+    with pytest.raises(JSONSchemaCompatibilityException):
+        obj.start(v1, 'first')
+
+
+def test_special_field_pass():
+    v1 = {
+        "type": "object",
+        "oneOf": [{
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "string"}
+            }
+        }, {
+            "type": "object",
+            "properties": {
+                "field_B": {"type": "integer"}
+            }
+        }]
+    }
+    obj = JSONSchemaValidation()
+    obj.start(v1, 'first')
+
+
+def test_success_two_schemas():
+    v1 = {
+        "type": "object",
+        "properties": {
+            "field_A": {"type": "string"}
+        },
+        "required": ["field_A"]
+    }
+    v2 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"}
+        },
+        "required": ["field_B"]
+    }
+    obj = JSONSchemaValidation()
+    obj.start(v1, 'first')
+    obj.start(v2, 'second')
+
+
+def test_failure_between_schemas():
+    v1 = {
+        "type": "object",
+        "properties": {
+            "field_A": {"type": "string"}
+        },
+        "required": ["field_A"]
+    }
+    v2 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"}
+        },
+        "required": ["field_B"]
+    }
+    v3 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"},
+            "field_A": {"type": "number"}
+        },
+        "required": ["field_B"]
+    }
+    obj = JSONSchemaValidation()
+    with pytest.raises(JSONSchemaCompatibilityException):
```

Comment:
Only the third one should raise the exception. The two first validation can be moved outside of this context.